pkg/metrics/record.go

```
@@ -0,0 +1,15 @@
+package metrics
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+)
+
+func Record(ctx context.Context, ms stats.Measurement, ros ...stats.Options) {
+	ros = append(ros, stats.WithMeasurements(ms))
+
+	if err := stats.RecordWithOptions(ctx, ros...); err != nil {
+		log.Error(err, "failed to record metric")
```

why not return the err here so you can bubble it up